koku/reporting/migrations/0303_tags_data.py

```diff
@@ -0,0 +1,51 @@
+import functools
+
+from django.db import migrations
+
+
+def migrate_tags(apps, schema_editor, provider):
+    EnabledTagKeys = apps.get_model("reporting", "EnabledTagKeys")
+    ProviderEnabledTagKeys = apps.get_model("reporting", f"{provider}EnabledTagKeys")
+    db_alias = schema_editor.connection.alias
+
+    current_records = ProviderEnabledTagKeys.objects.using(db_alias).all()
+    new_records = (
+        EnabledTagKeys(key=record.key, enabled=record.enabled, provider_type=provider) for record in current_records
+    )
+
+    EnabledTagKeys.objects.using(db_alias).bulk_create(new_records, ignore_conflicts=True)
+
+
+def unmigrate_tags(apps, schema_editor, provider):
+    EnabledTagKeys = apps.get_model("reporting", "EnabledTagKeys")
+    db_alias = schema_editor.connection.alias
+
+    EnabledTagKeys.objects.using(db_alias).filter(provider_type__exact=provider).delete()
```

Comment:
Do we need this? The reverse-migration should be a truncate table, or just a no-op. Why delete individual provider types from the table?

Reply:
@samdoran took lead on making sure that this migration could be rerun if needed. Although this may look silly, what we want to accomplish is to reinsert the tags IF there is a problem. Right now our default is "on conflict do nothing". So this delete is to capture any state changes between the first & second run of the migration. With the steps we have taken to turn off the download workers and ddf, this may be non trivial, but I want to let @samdoran provide his thoughts too.

Reply:
So, what I am advocating for is just deleting everything in the table if we need to roll this back. That is the outcome from a reverse migration _anyway_, so instead of filtering and deleting, we could just truncate the table on a rollback. That should be faster and less error prone.

Reply:
I originally had it just deleting everything in the table but wasn't sure if we needed the ability to selectively delete keys by provider. I also wasn't sure how to have one single rollback operation for all of the data migrations. Since I was calling `unmigrate_tags` for every single provider doing the same thing, I made it accept a provider param. I could do some clever things in the rollback function to ensure it only runs once if called multiple times, but maybe there is an official way to do that.

Reply:
This what you are after:
https://www.postgresql.org/docs/current/sql-truncate.html

`TRUNCATE TABLE {schema}.reporting_enabledtagkeys`

We can call this every time and there won't be a performance hit. But, 🤷🏻 , we shouldn't have to roll back the migration anyway.

Reply:
I pushed up a change. Hopefully we won't have to use it. 🤞